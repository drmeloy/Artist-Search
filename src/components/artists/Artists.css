@import url('https://fonts.googleapis.com/css?family=Raleway:300&display=swap');
@import url('https://fonts.googleapis.com/css?family=Raleway:800&display=swap');
@import url('https://fonts.googleapis.com/css?family=Montserrat:300&display=swap');

.Artists {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  align-items: center;
  overflow: auto;
  font-size: 30px;
  padding-bottom: 50px;
  margin-bottom: 0;

  li {
    list-style: none;
    display: inline-flex;
    justify-content: center;
    align-items: center;
    margin: 30px;
    width: 20vw;
    height: 20vh;
    border: 1px solid;
  }

  a {
    width: 100%;
    height: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    padding: 20px 60px;
  }

  p {
    margin: 0;
    text-align: center;
    height: 30%;
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    font-family: 'Montserrat', sans-serif;
    font-weight: 800;  
    text-decoration: none;
    color: white;
  }
}

/* H1 and Headers get Raleway Heavy, font-weight: 800 */
.Header {
  font-family: 'Raleway', sans-serif;
  font-weight: 800;
}


/* Sub-Headings get Raleway, font-weight 300 */
.subHeader {
  font-family: 'Raleway', sans-serif;
  font-weight: 300;
}


/* Any body text gets Montserrat, font-weight 300 */
.bodyText {
  font-family: 'Montserrat', sans-serif;
  font-weight: 300;
}
