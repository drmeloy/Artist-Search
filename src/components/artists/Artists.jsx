import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from './Artists.css';

const Artists = ({ artists }) => {
  const artistList = artists.map(artist => (
    <li key={artist.id} >
      <Link to={`/${artist.name}/${artist.id}`}>
        <p>{artist.name}</p>
        <p>{artist.disambiguation}</p>
      </Link>
    </li>
  ));

  return (
    <ul className={styles.Artists}>
      {artistList}
    </ul>
  );
};

Artists.propTypes = {
  artists: PropTypes.array.isRequired
};

export default Artists;
