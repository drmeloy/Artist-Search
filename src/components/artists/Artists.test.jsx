import React from 'react';
import { shallow } from 'enzyme';
import Artists from './Artists';

describe('Artists component', () => {
  it('matches its snapshot', () => {
    const wrapper = shallow(<Artists artists={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
