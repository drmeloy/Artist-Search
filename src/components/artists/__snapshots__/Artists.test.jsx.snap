// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artists component matches its snapshot 1`] = `<ul />`;
